docs/deployment-patterns/deployment-as-code.md

```diff
@@ -0,0 +1,113 @@
+
+Octopus has a comprehensive HTTP API and .NET SDK you can use to automate **everything** in Octopus - if you can do something through the user interface, you can automate it with code. You can create and update projects, variables, deployment processes, and more. Today, this is our only fully-supported **Octopus as Code** solution.
+
+### Future: Terraform provider {#terraform}
+
+There is an [open source Terraform provider for Octopus](https://github.com/MattHodge/terraform-provider-octopusdeploy), which is built on top of the Octopus HTTP API. We are using this Terraform provider ourselves for [Octopus Cloud](https://octopus.com/cloud) (our SaaS product), and we are actively contributing to it. It doesn’t cover 100% of all Octopus features yet, and the API surface (HCL) is subject to change. We will be building first-class support for this into the Octopus ecosystem in the future.
+
+If you want to do **Octopus configuration as code** today, we recommend using our .NET SDK which will always be supported. The Terraform provider will be a simpler, more declarative approach that we will support in the future.
+
+### Getting started
+
+Follow this process to move your Octopus project configuration to code:
+
+1. Move your [deployment scripts to code](#scripts-as-code) and decide if you want to take this next step
+1. Convert your project configuration to code
+  a. Target the HTTP API using your preferred method - we recommend the [.NET SDK](#dotnet-sdk)
+  b. Become an early adopter of the [Octopus Terraform provider](#terraform)
+1. Test your project configuration as code, targeting a dummy project so your real project continues to work uninterrupted
+1. Once you are happy with your testing, update your build pipeline to target the real project
+1. Train your people to make changes safely in this brave new world of **deployment as code**
+
+The process flow using **project as code** looks similar to what you already have, with on additional step to push the configuration changes into Octopus. Here is a typical example:
+
+1. Build and test application source code
+1. Package application source code
+1. **Push configuration changes to Octopus project treating the code as a source of truth**
+1. Push application packages to deployment feed
+1. Create a release in Octopus
+1. Deploy the release via Octopus to your Dev/Test environment
+1. Promote the release to other environments
+
+## Consistency and repeatability
+
+When you move to manage your Octopus configuration as code, Octopus still makes sure your deployments are consistent and repeatable. Whenever you push a configuration change to your Octopus project via code, it's just like people using the user interface or API to make changes. When you create a release, Octopus takes a snapshot of the deployment process, variables, and packages making every deployment of that release consistent and repeatable, regardless of whether the project was configured by a person or by code.
```

Comment:
I'm being fairly pedantic, but "When you move to manage" is a mouthful, I'd just say "When you manage"